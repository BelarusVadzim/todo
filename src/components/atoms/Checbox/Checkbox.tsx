import { PropsWithChildren } from "react";
import style from "./Checkbox.module.scss";

type CheckboxProps = PropsWithChildren;

const Label: React.FC<CheckboxProps> = ({ children }) => {
  return <span className={style.checkbox}>{children}</span>;
};

export default Label;
