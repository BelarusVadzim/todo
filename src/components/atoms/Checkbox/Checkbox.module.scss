@import 'src/variables.scss';

.checkbox_base{
  cursor: pointer;
  font-size: $font-size !important;
  //border: black solid 1px;
  // max-height: $font-size;
}

.checkbox_colored {
  background: -webkit-linear-gradient($primary-color, #e058ec);
  -webkit-background-clip: text;
          background-clip: text;
  -webkit-text-fill-color: transparent;
}

.checkbox_checked {
  @extend .checkbox_base;
  @extend .checkbox_colored;
}

.checkbox_unchecked {
  @extend .checkbox_base;
  color: $inactive-color;

  &:hover {
    @extend .checkbox_colored;
    }
}
