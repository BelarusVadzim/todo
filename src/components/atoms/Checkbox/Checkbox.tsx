import { useEffect, useReducer } from "react";
import style from "./Checkbox.module.scss";

type CheckboxProps = {
  checked?: boolean;
  onToggle?: (value: boolean) => void;
};

const Label: React.FC<CheckboxProps> = ({ checked = false, onToggle }) => {
  const [isChecked, toggle] = useReducer((isChecked) => !isChecked, checked);

  useEffect(() => onToggle && onToggle(isChecked), [isChecked, onToggle]);

  const materialIconName = isChecked
    ? "check_circle"
    : "radio_button_unchecked";

  const className = isChecked
    ? style.checkbox_checked
    : style.checkbox_unchecked;

  return (
    <span className={`material-icons ${className}`} onClick={toggle}>
      {materialIconName}
    </span>
  );
};

export default Label;
