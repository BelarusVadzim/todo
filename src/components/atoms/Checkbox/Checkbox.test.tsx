import render  from 'react-test-renderer';
import Checkbox from './Checkbox';

jest.mock('./Checkbox.module.scss', () => ({ checkboxChecked: 'checkboxCheckedStyle', checkboxUnchecked: 'checkboxUncheckedStyle' }));

describe('<Checkbox />', () => {
  it('should render properly', () => {
    const tree = render
      .create(<Checkbox />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when checked and classname specified should render properly ', () => {
    const tree = render
      .create(<Checkbox checked className='extClassName' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
