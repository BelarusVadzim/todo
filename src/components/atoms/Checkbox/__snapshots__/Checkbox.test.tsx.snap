// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render properly 1`] = `
<div
  className="material-icons checkboxUncheckedStyle"
>
  radio_button_unchecked
</div>
`;

exports[`<Checkbox /> when checked and classname specified should render properly  1`] = `
<div
  className="material-icons checkboxCheckedStyle extClassName"
>
  check_circle
</div>
`;
